Firestore/core/src/firebase/firestore/auth/token.cc

```diff
@@ -21,15 +21,15 @@ namespace firestore {
 namespace auth {
 
 Token::Token(const absl::string_view token, const User& user)
-    : token_(token), user_(user), is_valid_(true) {
+    : token_(token), user_(user) {
 }
 
-Token::Token() : token_(), user_(User::Unauthenticated()), is_valid_(false) {
+Token::Token() : token_(), user_(User::Unauthenticated()) {
 }
 
-const Token& Token::Invalid() {
-  static const Token kInvalidToken;
-  return kInvalidToken;
+const Token& Token::Unauthenticated() {
```

Comment:
`Invalid()` is added for `nil`. Now we don't need it. Do we want to add `Unauthenticated()`?

Reply:
Previously, "invalid" meant either: (a) error condition, or (b) nil.
Now, "unauthenticated" means only (b) nil.

So really, you could view this PR as:
1) a rename from invalid -> unauthenticated
2) remove error conditions from this state (since we can now track that via Status/StatusOr)